pam_tcb/pam_unix_auth.c

```diff
@@ -128,7 +128,7 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags,
 		pam_syslog(pamh, retval == PAM_SUCCESS ? LOG_INFO : LOG_NOTICE,
 		    "Authentication %s for %s from %s(uid=%u)",
 		    retval == PAM_SUCCESS ? "passed" : "failed", user,
-		    getlogin() ?: "", getuid());
+		    getlogin() ? getlogin() : "", getuid());
```

Comment:
I don't think the compiler would optimize these two `getlogin()` invocations, so if you really want to fight this GNU extension, please make sure `getlogin()` is not invoked twice.

Reply:
I think we should continue using the GNU extension, unless there's a very good reason not to. This code isn't currently meant to be too portable, so no reason to enable warnings forcing it to be more portable than necessary.

Reply:
I suppose this means omitting `-Wpedantic` option, which is not very useful anyway.

Reply:
I'll drop the commit and `-Wpedantic`then.

Reply:
Yes, no `-Wpedantic` and probably not fixing its other warnings as well.

Reply:
I've added a small `static inline`wrapper in `support.h` to solve this.